### ISSUE TITLE
[BUG] npx node@12 executes node@16

### ISSUE THREAD

**Issue by Kreijstal** [npm/cli#4204]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Documentation says you can specify node version with npx, but right now, when using `npx node@12` it executes latest version instead

### Expected Behavior

Execute latest version

### Steps To Reproduce

run `npx node@12` with node 16, install it, and then see the version.

### Environment

- npm:  7.24.0
- Node: v16.10.0
- OS: Windows 10 
- platform: Acer
- npm config:
```ini
; "builtin" config from C:\Users\k\scoop\apps\nodejs\16.10.0\node_modules\npm\npmrc

cache = "C:\\Users\\k\\scoop\\persist\\nodejs\\cache"
prefix = "C:\\Users\\k\\scoop\\persist\\nodejs\\bin"

; node bin location = C:\Users\k\scoop\apps\nodejs\current\node.exe
; cwd = C:\Users\k\ok\node-ffi-test
; HOME = C:\Users\k
; Run `npm config ls -l` to show all defaults.
```

**Comment by ljharb** [comment]
So if you run `npx node@12 --version` what do you get?

**Comment by ljharb** [comment]
(not that the node package on npm is a good way to use different versions of node)

**Comment by Kreijstal** [comment]
>npx node@12 --version
v16.10.0

**Comment by ljharb** [comment]
You’re on npm 7, can you try on npm 8?

**Comment by Kreijstal** [comment]
updated node and npm versions

```
C:\Users\k\ok\node-ffi-test>npm -v
8.3.0

C:\Users\k\ok\node-ffi-test>npx node@12 --version
v17.3.0
 ```

**Comment by wraithgar** [comment]
This should be fixed as of `npm@8.16.0`

```sh
~/D/n/s/npx $ node --version
v18.6.0
~/D/n/s/npx $ npx node@12 --version
Need to install the following packages:
  node@12.22.12
Ok to proceed? (y) 
v12.22.12
```
